File: food-delivery-backend/models/restaurant.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Restaurant struct {
	ID           primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Name         string             `bson:"name" json:"name"`
	Image        string             `bson:"image" json:"image"`
	Rating       float64            `bson:"rating" json:"rating"`
	DeliveryTime int32              `bson:"deliveryTime" json:"deliveryTime"`
	CuisineType  string             `bson:"cuisineType" json:"category"`
}

File: food_delivery_ui/lib/models/restaurant.dart
class Restaurant {
  final String id;
  final String name;
  final String image;
  final double rating;
  final int deliveryTime;
  final String category;

  Restaurant({
    required this.id,
    required this.name,
    required this.image,
    required this.rating,
    required this.deliveryTime,
    required this.category,
  });

  factory Restaurant.fromJson(Map<String,dynamic> json){
    return Restaurant(
      id: json["id"],
      name: json["name"],
      image: json["image"],
      rating: json["rating"].toDouble(),
      deliveryTime: json["deliveryTime"],
      category: json["category"],
    );
  }
}